gen/cycle.ml

```diff
@@ -702,24 +705,24 @@ let remove_store n0 =
 (* Set locations of events *)
 (***************************)
 
-  let is_non_fetch_and_same e =
-    is_real_edge e && same_loc e && not (E.is_fetch e)
+  let is_read_same_nonfetch n m =
+    n.evt.loc = m.evt.loc && n.evt.dir = Some R && not (E.is_ifetch n.edge.E.a1)
 
   let check_fetch n0 =
     let rec do_rec m =
       let p = find_real_edge_prev m.prev in
-      if E.is_fetch m.edge then begin
-        if E.is_fetch p.edge && find_real_edge_prev p.prev != m then
-          Warn.user_error "Bad consecutive fetches [%s] => [%s]"
-            (str_node p) (str_node m)
-      end ;
-      if
-        E.is_fetch p.edge && is_non_fetch_and_same m.edge ||
-        E.is_fetch m.edge && is_non_fetch_and_same p.edge
-      then begin
-        Warn.user_error "Ambiguous Data/Code location es [%s] => [%s]"
-          (str_node p) (str_node m)
-      end ;
+      (* ensure Instr read is followed or preceded by plain read to same location*)
+      if E.is_ifetch m.edge.E.a1 && m.evt.dir = Some R && not
+        (is_read_same_nonfetch m.next m ||
```

Comment:
Isn't this check redundant? I think find_node_prev (in the next line) will get to m.next, won't it?